File: src/main/java/uss/code/auth/dto/response/StudentIdAvailabilityResponse.java
package uss.code.auth.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;

import static lombok.AccessLevel.PRIVATE;

@Builder(access = PRIVATE)
public record StudentIdAvailabilityResponse(
        @Schema(
                description = "학번 사용 가능 여부",
                example = "true"
        )
        boolean available
) {
    public static StudentIdAvailabilityResponse of(final boolean available) {
        return StudentIdAvailabilityResponse.builder()
                .available(available)
                .build();
    }
}

File: src/main/kotlin/uss/code/auth/dto/response/StudentIdAvailabilityResponse.kt
package uss.code.auth.dto.response

import io.swagger.v3.oas.annotations.media.Schema

@JvmRecord
data class StudentIdAvailabilityResponse(
    @field:Schema(
        description = "학번 사용 가능 여부",
        example = "true"
    )
    val available: Boolean,
) {
    companion object {
        @JvmStatic
        fun of(available: Boolean): StudentIdAvailabilityResponse {
            return StudentIdAvailabilityResponse(available = available)
        }
    }
}
